packages/vscode-extension/src/connectionService.ts

```diff
@@ -286,15 +288,59 @@ export async function establishPersistentConnectionToSchemaPoller(
   );
 
   result.success
-    ? attachSchemaPollerConnectionErrorEventListener()
+    ? attachSchemaPollerConnectionEventListeners()
     : attachSchemaPollerConnectionFailedEventListeners();
 
   return result;
 }
 
-export function getConnectionDatabases(): Database[] {
+/**
+ *  Gets an array of database names from the connection, if they exist.
+ * @returns An array of database names, or an empty array if no databases exist.
+ */
+export function getConnectionDatabases(): Pick<
+  Database,
+  'name' | 'default' | 'home'
+>[] {
   const schemaPoller = getSchemaPoller();
-  return schemaPoller.connection?.databases ?? [];
+  const databases = schemaPoller.connection?.databases ?? [];
+
+  if (!schemaPoller.metadata || !schemaPoller.metadata.dbSchema) {
+    return databases;
+  }
+
+  return schemaPoller.metadata.dbSchema.databaseNames?.map((name) => {
+    const database = databases.find((db) => db.name === name);
+    if (!database) {
+      return {
+        name: name,
+        default: false,
+        home: false,
+      };
+    } else {
+      return database;
+    }
+  }, []);
```

Comment:
This ensures that we maintain an up to date list of databases in the connection tree view.

Because the schema poller only returns a list of database names, we need to map these to the actual `Database` objects in our connection. 

If a database is not found by name, we can can construct a new `Database` object that our tree view can consume